src/wmtk/invariants/MinIncidentValenceInvariant.cpp

```diff
@@ -0,0 +1,37 @@
+#include "MinIncidentValenceInvariant.hpp"
+
+#include <wmtk/Mesh.hpp>
+#include <wmtk/SimplicialComplex.hpp>
+
+namespace wmtk::invariants {
+
+
+MinIncidentValenceInvariant::MinIncidentValenceInvariant(const Mesh& m, long min_valence)
+    : MeshInvariant(m)
+    , m_min_valence(min_valence)
+{}
+
+bool MinIncidentValenceInvariant::before(const Tuple& t) const
+{
+    const Tuple v0 = t;
+    const Tuple v1 = mesh().switch_vertex(t);
+    const long val0 = static_cast<long>(SimplicialComplex::vertex_one_ring(mesh(), v0).size());
+    const long val1 = static_cast<long>(SimplicialComplex::vertex_one_ring(mesh(), v1).size());
+
+    return val0 >= m_min_valence && val1 >= m_min_valence;
+}
+
+bool MinIncidentValenceInvariant::after(PrimitiveType type, const std::vector<Tuple>& t) const
+{
+    assert(type == PrimitiveType::Edge);
+
+    for (const Tuple& e : t) {
+        if (!before(e)) {
```

Comment:
if we're going to call before in after might as well specify another function like `valence_improved(Tuple)` and call that. Then the semantic is that we are doing a type of check in both the before and after, rather than that we are checking preconditions in postdoncditiosn